packages/core/src/components/templates/PeriodInput/PeriodInput.tsx

```diff
@@ -64,6 +73,16 @@ const PeriodInput = ({
     setEndDate(endDateConfig.date);
   }, [endDateConfig.date]);
 
+  useEffect(() => {
+    if (interval) {
+      if (startDate && !endDate) {
+        setEndDate(new Date(startDate.getTime() + interval));
+      } else if (endDate && !startDate) {
+        setStartDate(new Date(endDate.getTime() - interval));
+      }
+    }
```

Comment:
By doing this this way, the value outside is not updated. You are only updating the visual part but endDate is still empty where the component is used.